code/modules/projectiles/projectile.dm

```diff
@@ -97,6 +97,14 @@
 				splatter_dir = get_dir(starting, target_loca)
 			if(isalien(L))
 				new /obj/effect/temp_visual/dir_setting/bloodsplatter/xenosplatter(target_loca, splatter_dir)
+			var/obj/item/bodypart/B = L.get_bodypart(def_zone)
+			if(B.status == BODYPART_ROBOTIC) // So if you hit a robotic, it sparks instead of bloodspatters

```

Comment:
Honestly, this whole thing should be a bodypart proc.

`/obj/item/bodypart/proc/make_blood()`

Then redefined for aliums and beep boops.
Will also work with bullet hits, I'm pretty sure.

Reply:
I'm not sure I agree:

Projectile code has to do a few calculations to check where the bloodspatter is coming from/going, as well as the location to drop the blood spatter. It also takes aliens into account. (obviously.)

Melee is just a chance to drop splatter when getting hit by sharp weapons.

Offloading just the bloodsplatter/oilsplatter into a proc seems like it's just moving code around without any particular gains, but maybe I'm just not understanding what you want as a part of it?